src/solvers/strings/string_refinement_util.h

```diff
@@ -133,6 +133,28 @@ class interval_sparse_arrayt final : public sparse_arrayt
   /// Complexity is logarithmic in the number of entries.
   exprt at(std::size_t index) const;
 
+  /// Iterators over interval boundaries
+  typedef std::map<std::size_t, exprt>::const_iterator const_iterator; // NOLINT
+  const_iterator begin() const
```

Comment:
I think these iterators can be confusing since one could think we are iterating over all indices rather than intervals.
By making `restrict_sparse_array_to_indices` a method of this class (or sparse_array), I think you wouldn't have to define them.